src_features/signMessageEIP712/path.c

```diff
@@ -285,23 +286,33 @@ static bool array_depth_list_pop(void) {
  * Updates the path so that it doesn't point to a struct-type field, but rather
  * only to actual fields.
  *
+ * @param[in] skip_if_array skip if path is already pointing at an array
+ * @param[in] stop_at_array stop at the first downstream array
  * @return whether the path update worked or not
  */
-static bool path_update(void) {
+static bool path_update(bool skip_if_array, bool stop_at_array) {
     uint8_t fields_count;
     const void *struct_ptr;
-    const void *field_ptr;
+    const void *starting_field_ptr, *field_ptr;
     const char *typename;
     uint8_t typename_len;
     uint8_t hash[KECCAK256_HASH_BYTESIZE];
 
     if (path_struct == NULL) {
         return false;
     }
-    if ((field_ptr = get_field(NULL)) == NULL) {
+    if ((starting_field_ptr = get_field(NULL)) == NULL) {
         return false;
     }
+    field_ptr = starting_field_ptr;
     while (struct_field_type(field_ptr) == TYPE_CUSTOM) {
```

Comment:
add a comment to explain what this loop is looking for

Reply:
Already explained in the function's comment :+1: